utils/runtime-fuzzer/src/gear_calls.rs

```diff
@@ -424,32 +421,35 @@ fn config(
         .into_iter(),
     );
 
-    let mut params_config = SyscallsParamsConfig::default();
-    params_config.add_rule(
-        ParamType::Regular(RegularParamType::Alloc),
-        (10..=20).into(),
-    );
-    params_config.add_rule(
-        ParamType::Regular(RegularParamType::Free),
+    let mut params_config = SyscallsParamsConfig::default_regular();
+    params_config.set_rule(RegularParamType::Alloc, (10..=20).into());
+    params_config.set_rule(
+        RegularParamType::Free,
         (initial_pages..=initial_pages + 35).into(),
     );
+    params_config.set_ptr_rule(PtrParamAllowedValues::Value(0..=0));
 
-    let existing_addresses = NonEmpty::collect(
+    let syscall_destination = NonEmpty::collect(
         programs
             .iter()
             .copied()
-            .filter(|&pid| pid != ProgramId::default()),
-    );
+            .filter(|&pid| pid != ProgramId::default())
+            .map(|pid| pid.into()),
+    )
+    .map(ActorKind::ExistingAddresses)
+    .unwrap_or(ActorKind::Source);
 
-    log::trace!(
-        "Messages will be sent to existing addresses {:?}",
-        existing_addresses
-    );
+    log::trace!("Messages destination config: {:?}", syscall_destination);
+
+    params_config.set_ptr_rule(PtrParamAllowedValues::ActorId(syscall_destination.clone()));
+    params_config.set_ptr_rule(PtrParamAllowedValues::ActorIdWithValue {
+        actor_kind: syscall_destination.clone(),
+        range: 0..=0,
```

Comment:
I don’t really understand whether value fuzzing works at all, since in tests you are using a non-empty range. Btw, also rename `syscall_destination`.

Reply:
done